### DISCUSSION TITLE
conf server的作用是什么

### DISCUSSION THREAD

**Discussion by sunyq** [apache/apisix#8525] [General]
在2.15和3.0版本的部署文档中，无论是traditional模式还是decoupled模式，都会有一个conf server
按照官方文档里的介绍：conf server充当APISIX和etcd之间的一层代理(listens on UNIX socket and acts as a proxy between APISIX and etcd)
请问为什么要在APISIX和etcd之间加一层conf server作为代理
看了下相关的配置和源码，这个conf server似乎没做什么实际的事情，就是一个请求转发？
为什么不让控制面和数据面直接去连etcd呢？

**Comment by tzssangglass** [comment]
in 3.0, if you deploy APISIX as DP, there is no conf server.

**Reply by sunyq** [reply]
But there is still a conf server in the CP instance. 
And the DP instances will fetch configuration from the conf server. 
So why design it this way?

**Reply by tzssangglass** [reply]
> And the DP instances will fetch configuration from the conf server.
> So why design it this way?

1. DP can use mTLS protocol to connect with conf_server. I remember that if DP uses mTLS to connect to etcd, there is a requirement for certificates.
2. conf_server can load balance requests (better than in lua-resty-etcd)

**Reply by sunyq** [reply]
cool, thanks!
